File: frontend/lib/app/model/workspace.dart
import 'package:equatable/equatable.dart';
import 'package:frontend/app/model/office.dart';

class Workspace extends Equatable {
  Workspace({
    required this.id,
    required this.name,
    required this.office,
  });

  String id;
  String name;
  Office office;

  factory Workspace.fromJson(Map<String, dynamic> json) => Workspace(
        id: json['id'] as String,
        name: json['name'] as String,
        office: Office.fromJson(json['officeID'] as Map<String, dynamic>),
      );

  Map<String, dynamic> toJson() {
    final map = <String, dynamic>{};
    map['id'] = id;
    map['name'] = name;
    map['officeID'] = office.toJson();
    return map;
  }

  /// return list of Workspaces from jsonList
  static List<Workspace> fromListDynamic(List<dynamic> jsonList) {
    final list = <Workspace>[];

    for (final element in jsonList) {
      list.add(Workspace.fromJson(element as Map<String, dynamic>));
    }
    return list;
  }

  @override
  List<Object?> get props => [name];
}

File: api/models/workspace.js
'use strict';
import mongoose from 'mongoose';
import office from './office';

const Schema = mongoose.Schema;
/*
    {
        name: "Workspace 1",
        officeID: '625689f1bc5bb6b2cdd790cf'
    }
 */
const workspaceSchema = new Schema({
    name: String,
    officeID: { type: mongoose.Types.ObjectId, ref: office, required: true},
});

export default mongoose.model('Workspace', workspaceSchema);